contracts/interchain-token-service/tests/execute.rs

```diff
@@ -314,6 +314,312 @@ fn execute_message_interchain_transfer_should_scale_the_amount_when_source_decim
     );
 }
 
+#[test]
+fn execute_message_interchain_transfer_should_scale_correctly_in_3_chain_cycle() {
+    let TestMessage {
+        source_its_contract,
+        ..
+    } = TestMessage::dummy();
+    let configs = vec![
+        (
+            "ethereum".parse().unwrap(),
+            source_its_contract.clone(),
+            Uint256::MAX.try_into().unwrap(),
+            18,
+        ),
+        (
+            "stellar".parse().unwrap(),
+            source_its_contract.clone(),
+            Uint256::from(u128::MAX).try_into().unwrap(),
+            12,
+        ),
+        (
+            "sui".parse().unwrap(),
+            source_its_contract.clone(),
+            Uint256::from(u64::MAX).try_into().unwrap(),
+            6,
```

Comment:
I think we can increase the coverage for this like below:

For each chain in the config, start a deployment of the token from it to other chains (token decimals can be set to chain's decimals maybe), and then do a transfer cycle that ends on the same chain.

This will test various scaling combinations:
18 -> 12 -> 6 -> 18
12 -> 6 -> 18 -> 12
6 -> 18 -> 12 -> 6

Generalizing this also makes it easier to add any chain config in this list in the future

Reply:
I will implement this in a follow up, want to get this test in for the release